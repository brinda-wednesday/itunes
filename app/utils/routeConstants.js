export default {
  itunes: {
    route: '/',
    exact: true
  },
  trackDetails: {
    route: '/:trackId',
    exact: true
  },
  repos: {
    route: '/github/repo',
    exact: true
  }
};
