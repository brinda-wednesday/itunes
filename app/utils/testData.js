export const testItunesData = {
  resultCount: 5,
  results: [
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 268893121,
      collectionId: 1440649280,
      trackId: 1440649635,
      artistName: 'Tyga',
      collectionName: 'Careless World: Rise of the Last King',
      trackName: 'Faded (feat. Lil Wayne)',
      collectionCensoredName: 'Careless World: Rise of the Last King',
      trackCensoredName: 'Faded (feat. Lil Wayne)',
      artistViewUrl: 'https://music.apple.com/us/artist/tyga/268893121?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/album/faded-feat-lil-wayne/1440649280?i=1440649635&uo=4',
      trackViewUrl: 'https://music.apple.com/us/album/faded-feat-lil-wayne/1440649280?i=1440649635&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/9b/8a/cb/9b8acba2-949a-acf7-16e6-c1a869e7340a/mzaf_8420185230668042391.plus.aac.p.m4a',
      artworkUrl30:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/8d/af/2e/8daf2ed5-b5c1-9c53-17fc-a10d7a247121/source/30x30bb.jpg',
      artworkUrl60:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/8d/af/2e/8daf2ed5-b5c1-9c53-17fc-a10d7a247121/source/60x60bb.jpg',
      artworkUrl100:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/8d/af/2e/8daf2ed5-b5c1-9c53-17fc-a10d7a247121/source/100x100bb.jpg',
      collectionPrice: 13.99,
      trackPrice: 1.29,
      releaseDate: '2012-01-13T12:00:00Z',
      collectionExplicitness: 'explicit',
      trackExplicitness: 'explicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 23,
      trackNumber: 10,
      trackTimeMillis: 211413,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Hip-Hop/Rap',
      contentAdvisoryRating: 'Explicit',
      isStreamable: true
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 1062085272,
      collectionId: 1445140820,
      trackId: 1445141316,
      artistName: 'Alan Walker',
      collectionName: 'Different World',
      trackName: 'Faded',
      collectionCensoredName: 'Different World',
      trackCensoredName: 'Faded',
      artistViewUrl: 'https://music.apple.com/us/artist/alan-walker/1062085272?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/album/faded/1445140820?i=1445141316&uo=4',
      trackViewUrl: 'https://music.apple.com/us/album/faded/1445140820?i=1445141316&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/d2/c1/c4/d2c1c496-4001-d10f-bbf6-626eb5bcf2cf/mzaf_7717849628751219162.plus.aac.p.m4a',
      artworkUrl30:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/a3/1b/0b/a31b0b2b-c444-f9d8-52d2-471402b4f6cb/source/30x30bb.jpg',
      artworkUrl60:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/a3/1b/0b/a31b0b2b-c444-f9d8-52d2-471402b4f6cb/source/60x60bb.jpg',
      artworkUrl100:
        'https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/a3/1b/0b/a31b0b2b-c444-f9d8-52d2-471402b4f6cb/source/100x100bb.jpg',
      collectionPrice: 9.99,
      trackPrice: 1.29,
      releaseDate: '2015-12-03T12:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 14,
      trackNumber: 15,
      trackTimeMillis: 212107,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Dance',
      isStreamable: true
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 261545033,
      collectionId: 918556383,
      trackId: 918556408,
      artistName: 'ZHU',
      collectionName: 'Faded (The Remixes) - EP',
      trackName: 'Faded',
      collectionCensoredName: 'Faded (The Remixes) - EP',
      trackCensoredName: 'Faded (ODESZA Remix)',
      artistViewUrl: 'https://music.apple.com/us/artist/zhu/261545033?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/album/faded-odesza-remix/918556383?i=918556408&uo=4',
      trackViewUrl: 'https://music.apple.com/us/album/faded-odesza-remix/918556383?i=918556408&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/36/16/36/36163607-50ce-5dec-846a-782467ea6b09/mzaf_529304218680119840.plus.aac.p.m4a',
      artworkUrl30:
        'https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/28/f4/e6/28f4e661-10c0-7d22-4eda-71c7f5e2ec29/source/30x30bb.jpg',
      artworkUrl60:
        'https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/28/f4/e6/28f4e661-10c0-7d22-4eda-71c7f5e2ec29/source/60x60bb.jpg',
      artworkUrl100:
        'https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/28/f4/e6/28f4e661-10c0-7d22-4eda-71c7f5e2ec29/source/100x100bb.jpg',
      collectionPrice: 4.99,
      trackPrice: 1.29,
      releaseDate: '2014-06-29T07:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 6,
      trackNumber: 1,
      trackTimeMillis: 238640,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Dance',
      isStreamable: true
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 1062085272,
      collectionId: 1196294554,
      trackId: 1196294581,
      artistName: 'Alan Walker',
      collectionName: 'Faded - EP',
      trackName: 'Faded',
      collectionCensoredName: 'Faded - EP',
      trackCensoredName: 'Faded',
      artistViewUrl: 'https://music.apple.com/us/artist/alan-walker/1062085272?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/album/faded/1196294554?i=1196294581&uo=4',
      trackViewUrl: 'https://music.apple.com/us/album/faded/1196294554?i=1196294581&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/f4/32/01/f43201b9-4bba-7654-2e43-d59e2d907e9f/mzaf_2440137894989713967.plus.aac.p.m4a',
      artworkUrl30:
        'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/34/15/a1/3415a1f7-68ee-b2c6-7b34-0ac3bb37e65a/source/30x30bb.jpg',
      artworkUrl60:
        'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/34/15/a1/3415a1f7-68ee-b2c6-7b34-0ac3bb37e65a/source/60x60bb.jpg',
      artworkUrl100:
        'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/34/15/a1/3415a1f7-68ee-b2c6-7b34-0ac3bb37e65a/source/100x100bb.jpg',
      collectionPrice: 3.99,
      trackPrice: 1.29,
      releaseDate: '2015-12-03T08:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 4,
      trackNumber: 1,
      trackTimeMillis: 212627,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Dance',
      isStreamable: true
    },
    {
      wrapperType: 'track',
      kind: 'song',
      artistId: 261545033,
      collectionId: 918184076,
      trackId: 918184083,
      artistName: 'ZHU',
      collectionName: 'THE NIGHTDAY - EP',
      trackName: 'Faded',
      collectionCensoredName: 'THE NIGHTDAY - EP',
      trackCensoredName: 'Faded',
      artistViewUrl: 'https://music.apple.com/us/artist/zhu/261545033?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/album/faded/918184076?i=918184083&uo=4',
      trackViewUrl: 'https://music.apple.com/us/album/faded/918184076?i=918184083&uo=4',
      previewUrl:
        'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/2c/3f/23/2c3f230d-9724-e12e-2801-6f3fc03a5ed1/mzaf_5483133236338116259.plus.aac.p.m4a',
      artworkUrl30:
        'https://is4-ssl.mzstatic.com/image/thumb/Music125/v4/15/80/ed/1580ed04-3681-df7d-7c3b-52068f16ccfc/source/30x30bb.jpg',
      artworkUrl60:
        'https://is4-ssl.mzstatic.com/image/thumb/Music125/v4/15/80/ed/1580ed04-3681-df7d-7c3b-52068f16ccfc/source/60x60bb.jpg',
      artworkUrl100:
        'https://is4-ssl.mzstatic.com/image/thumb/Music125/v4/15/80/ed/1580ed04-3681-df7d-7c3b-52068f16ccfc/source/100x100bb.jpg',
      collectionPrice: 4.99,
      trackPrice: 1.29,
      releaseDate: '2014-04-20T07:00:00Z',
      collectionExplicitness: 'notExplicit',
      trackExplicitness: 'notExplicit',
      discCount: 1,
      discNumber: 1,
      trackCount: 6,
      trackNumber: 2,
      trackTimeMillis: 223480,
      country: 'USA',
      currency: 'USD',
      primaryGenreName: 'Dance',
      isStreamable: true
    }
  ]
};

export const githubReposData = {
  totalCount: 3,
  items: [
    {
      fullName: 'qianguyihao/Mac',
      name: 'Mac',
      stargazersCount: 1546
    },
    {
      fullName: 'abc',
      name: 'abc',
      stargazersCount: 1345
    },
    {
      fullName: 'cdf',
      name: 'cdf',
      stargazersCount: 13
    }
  ]
};
