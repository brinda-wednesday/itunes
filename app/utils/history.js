import { createBrowserHistory } from 'history';
import routeConstants from '@utils/routeConstants';
const routes = Object.keys(routeConstants);
const pathname = window.location.pathname;
let baseUrl = '';
window.process = {
  env: {
    NODE_ENV: 'development'
  }
};
export function getHistory() {
  if (process.env.ENVIRONMENT_NAME === 'uat') {
    routes.forEach((routeKey) => {
      const route = routeConstants[routeKey].route;
      if (pathname.includes(route)) {
        if (pathname.substring(pathname.length - route.length, pathname.length) === route) {
          baseUrl = pathname.substring(0, pathname.length - route.length);
        }
        if (pathname.substring(pathname.length - route.length, pathname.length - 1) === `${route}/`) {
          baseUrl = pathname.substring(0, pathname.length - route.length - 1);
        }
      }
    });
  }
  return createBrowserHistory({ basename: baseUrl });
}

const history = getHistory();
export default history;
