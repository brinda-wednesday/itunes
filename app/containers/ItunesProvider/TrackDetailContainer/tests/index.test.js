/**
 *
 * Tests for TrackDetailContainer
 *
 *
 */

import React from 'react';
import { renderProvider } from '@utils/testUtils';
// import { fireEvent } from '@testing-library/dom'
import { TrackDetailContainerTest as TrackDetailContainer } from '../index';
import { testItunesData } from '@app/utils/testData';
import { BrowserRouter } from 'react-router-dom';

describe('<TrackDetailContainer /> container tests', () => {
  let submitSpy;

  beforeEach(() => {
    submitSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const data = testItunesData.results[0];
    const { baseElement } = renderProvider(
      <BrowserRouter>
        <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
      </BrowserRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
