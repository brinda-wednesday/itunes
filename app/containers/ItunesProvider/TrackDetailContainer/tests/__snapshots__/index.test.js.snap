// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetailContainer /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetailContainer__Container-sc-1wt099m-0 hPWKJQ"
    >
      <div
        class="ant-card ant-card-bordered TrackDetailContainer__CustomCard-sc-1wt099m-2 gEMjsW"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="example"
            class="TrackDetailContainer__Image-sc-1wt099m-1 cbBLqZ"
            src="https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/8d/af/2e/8daf2ed5-b5c1-9c53-17fc-a10d7a247121/source/100x100bb.jpg"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <h5
            class="ant-typography TrackDetailContainer__CustomTitle-sc-1wt099m-6 eUEWKz"
          >
            song
          </h5>
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Faded (feat. Lil Wayne)
              </div>
              <div
                class="ant-card-meta-description"
              >
                Tyga
              </div>
            </div>
          </div>
          <audio
            class="TrackDetailContainer__Audio-sc-1wt099m-3 ihqLXm"
            controls=""
          >
            <source
              class="TrackDetailContainer__Source-sc-1wt099m-4 eLriFX"
              data-testid="audio-src"
              src="https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/9b/8a/cb/9b8acba2-949a-acf7-16e6-c1a869e7340a/mzaf_8420185230668042391.plus.aac.p.m4a"
            />
          </audio>
          <span
            class="ant-typography TrackDetailContainer__CustomText-sc-1wt099m-5 kfUHax"
          >
             Collection Name : 
            Careless World: Rise of the Last King
          </span>
          <span
            class="ant-typography ant-typography-success TrackDetailContainer__CustomText-sc-1wt099m-5 kfUHax"
          >
            <i>
               
              Price : $
              1.29
            </i>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
